clippy_lints/src/manual_map.rs

```diff
@@ -182,13 +185,36 @@ impl LateLintPass<'_> for ManualMap {
                 expr.span,
                 "manual implementation of `Option::map`",
                 "try this",
-                format!("{}{}.map({})", scrutinee_str, as_ref_str, body_str),
+                if matches!(match_kind, MatchSource::IfLetDesugar { .. }) && is_else_of_if_let_else(cx, expr) {
+                    format!("{{ {}{}.map({}) }}", scrutinee_str, as_ref_str, body_str)
+                } else {
+                    format!("{}{}.map({})", scrutinee_str, as_ref_str, body_str)
+                },
                 app,
             );
         }
     }
 }
 
+fn is_else_of_if_let_else(cx: &LateContext<'_>, expr: &Expr<'_>) -> bool {
```

Comment:
move to utils (with docs)?

(Probably with all the other desugar functions)

Reply:
That was fixed as you were writing that. Forgot to commit that change.